plugins/lookup/output.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+name: output
+author: Polona Mihalič (@PolonaM)
+version_added: 1.0.0
+short_description: Read state file outputs.
+description:
+    - This lookup returns all outputs or selected output in state file.
+options:
+  _terms:
+    description:
+    - Names of the outputs to return.
+    - If value is set to C(all), all outputs will be returned.
+    required: True
+  project_path:
+    description:
+      - The path to the root of the Terraform directory with the terraform.tfstate file.
+    type: path
+  state_file:
+    description:
+      - Absolute path to an existing Terraform state file whose outputs will be listed.
+      - If this is not specified, the default C(terraform.tfstate) in the directory I(project_path) will be used.
+    type: path
+"""
+
+EXAMPLES = """
+- name: get selected output from terraform.tfstate
+  ansible.builtin.debug:
+    msg: "{{ lookup('cloud.terraform.output', 'my_output1', 'my_output2', project_path='path/to/project/dir/') }}"
+
+- name: get all outputs from custom state file
+  ansible.builtin.debug:
+    msg: "{{ lookup('cloud.terraform.output', 'all', state_file='path/to/custom/state/file') }}"
+"""
+
+RETURN = """
+_value:
+  description:
+    - Requested outputs.
+  type: list
+  elements: dict
+"""
+
+
+import os
+import json
+from ansible.errors import AnsibleError, AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+
+
+def state_file_path(project_path, state_file):
+    if project_path and os.path.exists("/".join((project_path, "terraform.tfstate"))):
+        return "/".join((project_path, "terraform.tfstate"))
+    elif state_file and os.path.exists(state_file):
+        return state_file
+    raise AnsibleLookupError(
+        'Could not find state_file "{0}", check the path and try again.'.format(
+            project_path or state_file
+        )
+    )
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        outputs = []
+        self.set_options(var_options=variables, direct=kwargs)
+        project_path = self.get_option("project_path")
+        state_file = self.get_option("state_file")
+
+        lookupfile = state_file_path(project_path, state_file)
+
+        try:
+            b_contents, show_data = self._loader._get_file_contents(lookupfile)
+            contents = json.loads(b_contents)
+            for term in terms:
+                if term == "all":
+                    for key, value in contents["outputs"].items():
+                        outputs.append({key: value})
```

Comment:
I suggest having two different output modes just like in the `terraform_output` module. If "everything" is specified, return a dict (not a list of single-element dicts!), otherwise just return the bare value. This will allow for usage like:

```yaml
- debug:
    msg: "my ip is {{ lookup('cloud.terraform.tf_output', 'my_machine_ip') }} and i am happy"
```

Reply:
Updated as suggested